src/worker.cc

```diff
@@ -512,21 +513,25 @@ void parse_locations(const rapidjson::Document& doc,
           // search_filter.exclude_ramp
           location->mutable_search_filter()->set_exclude_ramp(
               rapidjson::get_optional<bool>(*search_filter, "/exclude_ramp").get_value_or(false));
-          // search_filter.exclude_closures
-          auto exclude_closures = rapidjson::get_optional<bool>(*search_filter, "/exclude_closures");
-          // bail if you specified both of these, too confusing to work out how to use both at once
-          if (ignore_closures && exclude_closures) {
-            throw valhalla_exception_t{143};
-          }
-          // do we actually want to filter closures on THIS location
-          // NOTE: that ignore_closures takes precedence
-          location->mutable_search_filter()->set_exclude_closures(
-              ignore_closures ? !(*ignore_closures) : (exclude_closures ? *exclude_closures : true));
-          // set exclude_closures_disabled if any of the locations has the
-          // search_filter.exclude_closures set as false
-          if (!location->search_filter().exclude_closures()) {
-            exclude_closures_disabled = true;
-          }
+        }
+
+        // search_filter.exclude_closures must always be set because ignore_closures overrides it
+        // so if only ignore_closures is set we still need to set the search filter
```

Comment:
this block needed to be moved out of the if to make sure that exclude_closures is always properly set.